recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc-caam.sh

```diff
@@ -61,6 +67,30 @@ tdx_enc_key_gen() {
     fi
 }
 
+# backup original data in partition (if not encrypted)
+tdx_enc_backup_data() {
+    if [ ${TDX_ENC_PRESERVE_DATA} -eq 1 ]; then
+        mkdir -p "${TDX_ENC_STORAGE_MOUNTPOINT}"
+        if mount ${TDX_ENC_STORAGE_LOCATION} "${TDX_ENC_STORAGE_MOUNTPOINT}"; then
+            tdx_enc_log "Backing up original content..."
+
+            TDX_ENC_PRESERVE_LOCATION=$(mktemp)
+            BACKUP_STORAGE_LIMIT="$(awk '/MemFree/ { printf "%u", $2/30 }' /proc/meminfo)"
```

Comment:
Can we do this without `awk`? If not, you must add `awk` as a runtime dependency in the recipe.

Reply:
awk is part of BusyBox. I was wondering if I should add a cfg file for that to ensure it gets included? Would be the same for grep too.

Reply:
I only did it this way as it is much easier to read and less pipes to achieve the desired result. Other option was to use grep as in @sergioprado original comment:

> `# grep MemFree: /proc/meminfo | tr -s ' ' | cut -d ' ' -f 2`

However, that doesn't do the 30% calculation.

Reply:
I believe we want the feature to work also on Torizon OS images so minimizing the dependencies is a good thing. Depending on coreutils and grep should be okay.

The below code works on the busybox `sh` of the reference image:

```
BACKUP_STORAGE_PCT=30    # Configurable percentage.
...
MEM_FREE=$(grep MemFree: /proc/meminfo | tr -s ' ' | cut -d ' ' -f 2)
BACKUP_STORAGE_LIMIT=$((MEM_FREE * BACKUP_STORAGE_PCT / 100))
```

Reply:
Latest test seems to work with these changes. Have updated my branch so this should now be included in latest PR changes.